amd_openvx_extensions/amd_nn/src/topk_layer.cpp

```diff
@@ -0,0 +1,301 @@
+    ERROR_CHECK_STATUS(vxSetMetaFormatAttribute(metas[5], VX_TENSOR_DIMS, output_dims_1, sizeof(output_dims_1)));
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[6], VX_TENSOR_NUMBER_OF_DIMS, &num_dims, sizeof(num_dims)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[6], VX_TENSOR_DATA_TYPE, &type, sizeof(type)));
+    if (num_dims != 4) return VX_ERROR_INVALID_DIMENSION;
+    if (type != VX_TYPE_INT64) return ERRMSG(VX_ERROR_INVALID_TYPE, "validate: TopK: #7 output tensor data 'indices' type=%d (must be int64)\n", type);
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[6], VX_TENSOR_DIMS, output_dims_2, sizeof(output_dims_2)));
+
+    // output tensor configuration
+    ERROR_CHECK_STATUS(vxSetMetaFormatAttribute(metas[6], VX_TENSOR_DATA_TYPE, &type, sizeof(type)));
+    ERROR_CHECK_STATUS(vxSetMetaFormatAttribute(metas[6], VX_TENSOR_NUMBER_OF_DIMS, &num_dims, sizeof(num_dims)));
+    ERROR_CHECK_STATUS(vxSetMetaFormatAttribute(metas[6], VX_TENSOR_DIMS, output_dims_2, sizeof(output_dims_2)));
+
+    return VX_SUCCESS;
+}
+
+static vx_status VX_CALLBACK processTopKLayer(vx_node node, const vx_reference * parameters, vx_uint32 num)
+{
+    //get tensor dimensions
+    vx_size input_dims_0[4], input_dims_1[4], output_dims_0[4], output_dims_1[4];
+    vx_size num_of_dims;
+
+    vx_enum usage = VX_READ_ONLY;
+    vx_status status;
+    vx_map_id map_id;
+    vx_size stride[4];
+
+    //query and copy inputs
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, input_dims_0, sizeof(input_dims_0)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_NUMBER_OF_DIMS, &num_of_dims, sizeof(num_of_dims)));
+
+    float * ptr_input_0;
+    vx_size count_input_dims_0 = input_dims_0[0]*input_dims_0[1]*input_dims_0[2]*input_dims_0[3];
+    float *x_tensor = new float[count_input_dims_0];
+    //temporary moving ptr
+    float *x_tensor_temp = x_tensor;
+
+    status = vxMapTensorPatch((vx_tensor)parameters[0], num_of_dims, nullptr, nullptr, &map_id, stride, (void **)&ptr_input_0, usage, VX_MEMORY_TYPE_HOST, 0);
+    if(status)
+    {
+        std::cerr << "ERROR: vxMapTensorPatch() failed for input#1 (" << status << ")" << std::endl;
+        return -1;
+    }
+
+    memcpy(x_tensor_temp, ptr_input_0, (count_input_dims_0*sizeof(float)));
+ 
+
+    status = vxUnmapTensorPatch((vx_tensor)parameters[0], map_id);
+
+    if(status) 
+    {
+        std::cerr << "ERROR: vxUnmapTensorPatch() failed for input#5 (" << status << ")" << std::endl;
+        return -1;
+    }
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_DIMS, input_dims_1, sizeof(input_dims_1)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_NUMBER_OF_DIMS, &num_of_dims, sizeof(num_of_dims)));
+
+    
+    int64_t * ptr_input_1;
+    vx_size count_input_dims_1 = input_dims_1[0];
+    int64_t *k_tensor = new int64_t[count_input_dims_1];

```

Comment:
suggest using better name for k_tensor (tensor usually indicates vx_tensor object). What does k correspond(name)?
May be use name_tensor_data_buffer.

Reply:
It's called X and K in onnx. So I stuck to it